runtime/src/asset.rs

```diff
@@ -157,6 +179,70 @@ decl_module! {
         /// initialize the default event for this module
         fn deposit_event() = default;
 
+        /// This function is used to either register a new ticker or extend validity of an exisitng ticker
+        /// NB Ticker validity does not get carryforward when renewing ticker
+        ///
+        /// # Arguments
+        /// * `origin` It consist the signing key of the caller (i.e who signed the transaction to execute this function)
+        /// * `to_did` DID of the (future) owner of the ticker
+        /// * `_ticker` ticker to register
+        pub fn register_ticker(origin, to_did: IdentityId, _ticker: Vec<u8>) -> Result {
+            let _ = ensure_signed(origin)?;
+
+            let ticker = utils::bytes_to_upper(_ticker.as_slice());
```

Comment:
I've seen a lot of this, and sometime this is called from nested function.
A suggestion for a **Code Hygiene** task could be to *model* a Ticker structure, instead of use `Vec<u8>`. In that way we could be able to ensure *upper* case from its `fn new` or `From` trait.